package kingso_cm;

enum ResponseType
{
    CM_SUCCEED=0;
    CM_QUERY_FAILED=1;
    CM_QUERY_TIMEOUT=2;
};

enum NodeState
{
    Normal=0;     // 0 
    abnormal=1;   // 1
    timeout=2;    // 2
    unvalid=3;    // 3
    uninit=4;     // 4
};

message NodeInfo
{
    required string     node_ip=1;
    required uint32     node_port=2;
    required int32      protocol=3;
    required NodeState  state=4;   
    required int32      col_id=5;
};

message CMResponse
{
    required ResponseType   response_type=1;
    repeated NodeInfo       nodes=2;
    required int32          cluster_num=3;
};

